packages/scalecube-microservice/tests/integration/Errors/remoteServiceCall.spec.ts

```diff
@@ -0,0 +1,148 @@
+      receiverServiceDefinition: greetingServiceDefinition,
+      isRemote: true,
+    },
+  ])(
+    `
+        Given  'sender' and a 'receiverServiceDefinition' (receiver)
+               sender (microservice)       | receiverServiceDefinition (receiver)    |
+               microserviceWithServices    | greetingServiceDefinition               | # LocalCall
+               microserviceWithoutServices | greetingServiceDefinition               | # RemoteCallCall
+
+        `,
+    (connect) => {
+      const { sender, receiverServiceDefinition, isRemote } = connect;
+
+      // @ts-ignore
+      if (isRemote && global.isNodeEvn) {
+        return; // TODO: RFC - remoteCall nodejs
+      }
+
+      test(`
+    Scenario: service reject
+    Given     proxy and remoteService
+    When      remoteService reject
+    Then      proxy receive the error message
+
+  `, () => {
+        expect.assertions(1);
+        const proxy = sender.createProxy({ serviceDefinition: receiverServiceDefinition });
+
+        return expect(proxy.hello('Me')).rejects.toMatchObject(new Error(errorMessage));
+      });
+
+      test(`
+    Scenario: service emit error
+    Given     proxy and remoteService
+    When      remoteService emit error
+    Then      proxy receive the error message
+  `, (done) => {
+        expect.assertions(1);
+
+        const proxy = sender.createProxy({ serviceDefinition: receiverServiceDefinition });
+        proxy
+          .greet$(['Me'])
+          .pipe(
+            catchError((error: any) => {
+              expect(error).toMatchObject(new Error(errorMessage));
+              done();
+              return [];
+            })
+          )
+          .subscribe();
+      });
+
+      test(`
+    Scenario: service emit instead of resolve.
+    Given     proxy
+      And     remoteService
+              | method               | asyncModel definition | asyncModel reference|
+              | incorrectAsyncModel  | requestResponse       | requestStream       |
+    When      remoteService emit values
+    Then      proxy receive only first emit
+  `, () => {
+        expect.assertions(1);
+        const proxy = sender.createProxy({ serviceDefinition: receiverServiceDefinition });
+        return expect(proxy.incorrectAsyncModel('Me')).resolves.toMatchObject({ emptyMessage });
+      });
+
+      test(`
+    Scenario: service resolve instead of emit.
+    Given     proxy
+      And     remoteService
+              | method                | asyncModel definition | asyncModel reference|
+              | incorrectAsyncModel$  | requestStream         | requestResponse     |
+    When      remoteService return value
+    Then      proxy receive the value
+  `, (done) => {
+        expect.assertions(1);
+        const proxy = sender.createProxy({ serviceDefinition: receiverServiceDefinition });
+        proxy.incorrectAsyncModel$().subscribe((res: any) => {
+          expect(res).toMatchObject({ emptyMessage });
+          done();
```

Comment:
you don't have to call done here - test should wait for the emit, if you have specified the amount of expectations